### PR TITLE
Upgrade to Infinispan 9.0.0.Beta2

### PR DESCRIPTION
**Pull request by wburns**
Upgraded to latest Infinispan 9.0.0.Beta2.  This way when Final comes out there shouldn't be any required changes.  Note this also upgrades JGroups to the latest 4.0.0.Beta3.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
   <properties>
     <stack.version>3.4.0-SNAPSHOT</stack.version>
     <asciidoc.dir>${project.basedir}/src/main/asciidoc</asciidoc.dir>
-    <infinispan.version>8.2.5.Final</infinispan.version>
+    <infinispan.version>9.0.0.Beta1</infinispan.version>
```
Comment (tsegismont) [inline pom.xml]: It seems there's a `9.0.0.Beta2` already, can we upgrade to this one instead?

Comment (wburns) [inline pom.xml]: Sure there were some issues with resolving Beta2 in maven central so I was just playing it safe [1].  I will put in Beta2 though.

[1] http://lists.jboss.org/pipermail/infinispan-dev/2017-January/017149.html

**Comment by wburns** [comment]
Alright it should be all updated to Beta2.  Updated schema and removed default cache definition.  Also added a commit to ensure when used outside of a vert.x thread that the handler is always invoked on the vert.x context thread.

**Comment by tsegismont** [comment]
Thanks @wburns and congrats for your first Vert.x PR ;-)

**Pull request merged**